proposed/phpdoc.md

```diff
@@ -0,0 +1,2214 @@
+      // there should be no docblock here
+      $sqlite->open('/my/database/path');
+      <...>
+  }
+  ```
+
+* "DocComment" is a special type of comment which MUST
+
+  - start with the character sequence `/**` followed by a whitespace character
+  - end with `*/` and
+  - have zero or more lines in between.
+
+  In case a DocComment spans multiple lines then every line MUST start with
+  an asterisk (`*`) that SHOULD be aligned with the first asterisk of the
+  opening clause.
+
+  Single line example:
+
+  ```php
+  /** <...> */
+  ```
+
+  Multiline example:
+
+  ```php
+    /**
+     * <...>
+     */
+  ```
+
+* "DocBlock" is a "DocComment" containing a single "PHPDoc" structure and
+  represents the basic in-source representation.
+
+* "Tag" is a single piece of meta information regarding a "Structural Element"
+  or a component thereof.
+
+* "Inline PHPDoc" is a "PHPDoc" that is related to a "Tag" instead of a
+  "Structural element". It replaces the description part of the "Tag".
+
+* "Type" is the determination of what type of data is associated with an element.
+  This is commonly used when determining the exact values of arguments, constants,
+  properties and more.
+
+  See Appendix A for more detailed information about types.
+
+* "Semantic Version" refers to the definition as set in the [Semantic Versioning
+  Specification 2.0.0][SEMVER2].
+
+* "FQSEN" is an abbreviation for Fully Qualified Structural Element Name. This
+  notation expands on the Fully Qualified Class Name and adds a notation to
+  identify class/interface/trait members and re-apply the principles of the FQCN
+  to Interfaces, Traits, Functions and global Constants.
+
+  The following notations can be used per type of "Structural Element":
+
+  *Namespace*:      `\My\Space`
+  *Function*:       `\My\Space\myFunction()`
+  *Constant*:       `\My\Space\MY_CONSTANT`
+  *Class*:          `\My\Space\MyClass`
+  *Interface*:      `\My\Space\MyInterface`
+  *Trait*:          `\My\Space\MyTrait`
+  *Method*:         `\My\Space\MyClass::myMethod()`
+  *Property*:       `\My\Space\MyClass::$my_property`
+  *Class Constant*: `\My\Space\MyClass::MY_CONSTANT`
+
+  A FQSEN has the following [ABNF][RFC5234]
+  definition:
+
+          FQSEN    = fqnn / fqcn / constant / method / property  / function
+          fqnn     = "\" [name] *("\" [name])
+          fqcn     = fqnn "\" name
+          constant = (fqnn "\" / fqcn "::") name
+          method   = fqcn "::" name "()"
+          property = fqcn "::$" name
+          function = fqnn "\" name "()"
+          name     = (ALPHA / "_") *(ALPHA / DIGIT / "_")
+
+## 4. Basic Principles
+
+* A PHPDoc MUST always be contained in a "DocComment"; the combination of these
+  two is called a "DocBlock".
+
+* A DocBlock MUST directly precede a "Structural Element"
+
+  > An exception to this principle is the File-level DocBlock which MUST be
+  > placed at the top of a PHP source code file as the first DocBlock in a 
+  > file. 
```

Comment:
Is this implying that *every single file in a project* must have a file-level docblock?